In [1]:
# Consider the string given below and answer the following questions 
string = "OQYWFClFhFGAvIWYwGKpmZhnJiyzTslSIhSwvOsqJMEphzmifTkyqOMNpnOtXZxmCfgDYqbaBHAUvIWhMnvwZnEMVDvmEfLrDoQnAZgQEgXQVnmSYkfedpAdhrtpOgORpYLRZYGWdhWYuqQssCUXtTzKRDAhpjUheOzUroZNzWFtZOVwIapzUYtbSbjYNErzQ"
len(string)

193

####  Calculate the number of characters in the string

-  193
-  200
-  300
-  500


In [4]:
string.startswith('if')

False

#### Count the number of occurences of 'a' in the string

-  1
-  2
-  10
-  5

In [3]:
print(string.count('if'))

1


#### Identify whether the string starts with 'if'.

-  True
-  False

In [5]:
print(string[63:89])

mCfgDYqbaBHAUvIWhMnvwZnEMV


#### Return the substring starting at index 63 and ending at index 88  [Both the indexes are inclusive]

-  'OQYWFClFhFGA'
-  'mCfgDYqbaBHAUvIWhMnvwZnEMV'
-  'IWYwGKpmZhnJiyzTslSIhSwv'
-  'JMEphzmifTkyqOMNpnOtXZ'

In [10]:
string[45]

'z'

#### Identify the character at index 45

-  a
-  b
-  z
-  x

Practice Questions - I

In [11]:
number='159'

print(type(number))

<class 'str'>


In [12]:
number=34.721
print(float(int(number)))

34.0


In [13]:
name = Alex
print(type(name))

NameError: name 'Alex' is not defined

In [14]:
a = 8.9
b = 2
c = 5
print(a//b//c)

0.0


In [15]:
print(4 // 8 + ( 2.5 + 7.5 ) - 6 * 3)

-8.0


In [16]:
print(5 % ( 2 + 12 ) ** 3 - 60 // ( 7 + 1 ) * 2)

-9


In [17]:
string_1 = "I love to code in python"
print( string_1[7:13])

to cod


In [20]:
sentence = 'This sentence needs to be split into words'

print(sentence.split()[5])

split


In [21]:
len = 'python'
print(len(len))

TypeError: 'str' object is not callable

In [24]:
x=10
if x%2==0
print("even number")
else
print("odd number") 

SyntaxError: expected ':' (4058991420.py, line 2)

In [25]:
if 4+9<10:
  print("true")
else:
  print("false")
print("true")

false
true


In [27]:
x = 67
if x>50:
  if x>90:
    print("excellent")
  else:
    if x>80:
      print("good")
    else:
      print("fair")
else:
  if x>30:
    print("average")
  else:
    print("poor")

fair


In [29]:
a='pqrstuv'
i='j'
while i in a:
  print(i)

In [31]:
i = 16
while False:
    if i%2 == 0:
        break
    print(i)
    i += 2

In [32]:
s="Upgrad"

for i in range(len(s)):
    if s[i] in 'a':
        s[i]='*'

print(s)

TypeError: 'str' object is not callable